app/lib/actions/katello/content_view/incremental_updates.rb

```diff
@@ -29,14 +35,40 @@ def plan(version_environments, composite_version_environments, content, dep_solv
             end
 
             if hosts.any? && !content[:errata_ids].blank?
-              errata = ::Katello::Erratum.with_identifiers(content[:errata_ids])
-              hosts = hosts.where(:id => ::Katello::Host::ContentFacet.with_applicable_errata(errata).pluck(:host_id))
               plan_action(::Actions::BulkAction, ::Actions::Katello::Host::Erratum::ApplicableErrataInstall, hosts, content[:errata_ids])
             end
             plan_self(:version_outputs => output_for_version_ids)
           end
         end
 
+        def find_applicable_hosts_and_smart_proxies(hosts, content)
+          smart_proxies = []
+          if hosts.any? && !content[:errata_ids].blank?
+            errata = ::Katello::Erratum.with_identifiers(content[:errata_ids])
+            hosts = hosts.where(:id => ::Katello::Host::ContentFacet.with_applicable_errata(errata).pluck(:host_id))
+            smart_proxies = (hosts.map { |host| host.subscription_facet_attributes.registered_through unless host.subscription_facet_attributes.nil? }).uniq
+            smart_proxies = (smart_proxies.map { |proxy| SmartProxy.find_by_name(proxy) unless proxy.nil? || SmartProxy.find_by_name(proxy).default_capsule? }).compact
+          end
+          return hosts, smart_proxies
+        end
+
+        def plan_smart_proxy_sync(smart_proxies, environment, version)
+          if smart_proxies.any?
+            concurrence do
+              environment.each do |env|
+                if ::Katello::CapsuleContent.sync_needed?(env)
```

Comment:
CapsuleContent no longer exists in master, removed in: https://github.com/Katello/katello/pull/7750

Moved to the SmartProxy class

Reply:
but i think you are correct, this is redundant to the ::Katello::CapsuleContent.with_environment(env).map { |capsule| capsule.capsule }

Reply:
Thanks for confirmation, removing it.